payments-core/src/main/java/com/stripe/android/paymentsheet/viewmodels/BaseSheetViewModel.kt

```diff
@@ -179,7 +181,7 @@ internal abstract class BaseSheetViewModel<TransitionTargetType>(
     private fun fetchSavedSelection() {
         viewModelScope.launch {
             val savedSelection = withContext(workContext) {
-                prefsRepository.getSavedSelection()
+                prefsRepository.getSavedSelection(isGooglePayReady.asFlow().first())
```

Comment:
should this be a `isGooglePayReady.value` and error out if it's null? it might get updated multiple times

Reply:
We want to wait until it's fetched, it will initially be null but then become true or false. We already wait for it to be fetched before we call `createFragmentConfig()` which shows the fragment.
What do you mean by "it might get updated multiple times"? We only fetch the status once.

Reply:
oh ok, didn't realize we need to wait. nit: could you just inline `fetchSavedSelection` into `init{}` ? That prevents `fetchSavedSelection` gets incorrectly called multiple times, where the boolean might have a different value

Reply:
Done